pyfr/plugins/solutionwriter.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+
+from collections import OrderedDict
+
+from pyfr.h5writer import H5Writer
+from pyfr.inifile import Inifile
+from pyfr.plugins.base import BasePlugin
+
+
+class SolutionWriterPlugin(BasePlugin, H5Writer):
+    name = 'solutionwriter'
+    systems = ['*']
+
+    def __init__(self, intg, cfgsect, *args, **kwargs):
+        BasePlugin.__init__(self, intg, cfgsect)
+        H5Writer.__init__(self, intg, cfgsect)
+        self.dt = self.cfg.getfloat(self.cfgsect, 'dt', -1)
+        self.tprev = intg.tcurr - self.dt
```

Comment:
Working in terms of `tnext` as opposed to `tprev` will probably make your life easier.